### PR TITLE
feat: implement document analysis tool

### PR DESCRIPTION
**Pull request by hugohonda**

### CODE REVIEW

**Comment by hugohonda** [comment]
The output from generated code:

```json
{'document_metadata': {'title': 'MARKET COMMENTARY',
  'type': 'Financial Analysis Report',
  'confidentiality': 'CONFIDENTIAL - INTENDED FOR ORIGINAL RECIPIENTS ONLY'},
 'financial_data': {'chart_title': 'Total CapEx Spending',
  'metric': 'Total Cloud CapEx Spending',
  'unit': 'Billion USD',
  'yearly_data': [{'year': '2013', 'amount': '$28B'},
   {'year': '2014', 'amount': '$35B'},
   {'year': '2015', 'amount': '$39B'},
   {'year': '2016', 'amount': '$45B'},
   {'year': '2017', 'amount': '$52B'},
   {'year': '2018', 'amount': '$81B'},
   {'year': '2019', 'amount': '$75B'},
   {'year': '2020', 'amount': '$79B'},
   {'year': '2021', 'amount': '$95B'},
   {'year': '2022', 'amount': '$118B'},
   {'year': '2023', 'amount': '$120B'},
   {'year': '2024E', 'amount': '$182B'},
   {'year': '2025E', 'amount': '$196B'}],
  'projections': {'period': '2024E-2025E', 'cumulative_spending': '$378B'}},
 'key_insights': {'executive_commentary': {'speaker': 'Andy Jassy (Amazon CEO)',
   'key_points': ['Customer demand signals guide capacity planning',
    'Heavy investment in AI and infrastructure',
    'Demand exceeds current capacity']},
  'market_conditions': {'recent_volatility': {'S&P 500': '-3%',
    'Nasdaq': '-3.4%',
    'Nikkei': '-12.4%'},
   'outlook': 'Markets have largely recovered, but AI sector stocks remain below recent highs'}}}
```

Two things I observed when running the generated code:

1. The output gets really restricted to financial theme, probably related to the first analysis from vision agent:

image

2. The output structure is based on the docstring + vision agent interpretation, it is not a natural language response but another JSON

File: vision_agent/tools/tools.py [hunk 1]
```diff
@@ -1878,6 +1878,74 @@ def closest_box_distance(
...
+                                    'unit': 'Billion USD',
+                                    'xAxis': 'Year',
+                                    'yAxis': 'Total CapEx Spending'},
+                                'label': 'picture',
+                        '       summary': 'This bar chart illustrates the trend of '
+                                   'Total CapEx (Capital Expenditure) Spending '
+                                   'over time, from 2024 to 2025.'},
+
+                    (...)
+
```
Comment (dillonalaird) [inline vision_agent/tools/tools.py:1927]: I would make this a shorter, so try to keep the entire doc similar size to other tools. The longer the doc, the more likely the LLM is to choose it.

File: vision_agent/tools/tools.py [hunk 2]
```diff
@@ -1878,6 +1878,74 @@ def closest_box_distance(
...
+    }
+
+    response: dict[str, Any] = send_inference_request(
+        payload=payload,
+        endpoint_name="document-analysis",
+        files=files,
+        v2=True,
+        metadata_payload={"function_name": "document_analysis"},
+    )
+
```
Comment (dillonalaird) [inline vision_agent/tools/tools.py:1945]: One thing that just got added is this this function https://github.com/landing-ai/vision-agent/blob/main/vision_agent/tools/tools.py#L203

This is to re-display the data for the front end to use it. Could you add this as well?

**Review by dillonalaird — CHANGES_REQUESTED** [review]

File: vision_agent/tools/tools.py (hunk 1 above)
Comment (hugohonda) [reply vision_agent/tools/tools.py]: Made it shorter keeping the chart example, I can also remove it if it still too long

File: vision_agent/tools/tools.py (hunk 2 above)
Comment (hugohonda) [reply vision_agent/tools/tools.py:1928]: Ok, added! Thanks

**Review by dillonalaird — APPROVED** [review]
LGTM

**Pull request merged**